### PR TITLE
Arpit/och 662 Accounts and Counterparty API Wrapper

### PR DESCRIPTION
**Pull request by collegeimprovements**

### CODE REVIEW

File: lib/modern_treasurer/bank_account/account.ex [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  def create_external_account(
+        attrs \\ %{
+          body_params: %{
+            name: "test-exa-1",
+            counterparty_id: "bc665f7b-b08c-4ac3-9c18-ba926d44a60a"
+          }
+        }
+      ) do
+    body_params = Map.get(attrs, :body_params, %{})
+    # |> IO.inspect(label: "BODY PARAMS:")
```
Comment (jalbertsr) [inline lib/modern_treasurer/bank_account/account.ex:45]: we can remove that

File: lib/modern_treasurer/bank_account/account.ex [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  end
+
+  @doc """
+  `iex> ModernTreasurer.Account.create_external_account(%{name: "test-222", body_params: %{counterparty_id: "bc665f7b-b08c-4ac3-9c18-ba926d44a60a"}})`
+  """
+  def create_external_account(
+        attrs \\ %{
+          body_params: %{
+            name: "test-exa-1",
+            counterparty_id: "bc665f7b-b08c-4ac3-9c18-ba926d44a60a"
```
Comment (jalbertsr) [inline lib/modern_treasurer/bank_account/account.ex:40]: why these default parameters?

File: lib/modern_treasurer/bank_account/account.ex (hunk 1 above)
Comment (collegeimprovements) [reply lib/modern_treasurer/bank_account/account.ex:45]: Doing it

File: lib/modern_treasurer/bank_account/account.ex (hunk 2 above)
Comment (collegeimprovements) [reply lib/modern_treasurer/bank_account/account.ex:40]: Need to remove it. On it.

File: lib/modern_treasurer/counterparty/counter_party.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  APIs for CounterParty
+  """
+  alias ModernTreasurer.HTTP.Client
+
+  def list_counterparties(attrs) do
+    query_params = Map.get(attrs, :query_params, []) |> Enum.into([])
+    Client.get("/counterparties", query: query_params)
+  end
+
+  def create_counterparty(attrs \\ %{name: "test-cp-tesla-1"}) do
```
Comment (jalbertsr) [inline lib/modern_treasurer/counterparty/counter_party.ex:12]: not sure if it is a good practice to leave default value like that?

File: lib/modern_treasurer/counterparty/counter_party.ex [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    Client.post("/counterparties", attrs)
+  end
+
+  def get_counterparty(attrs \\ %{id: "bc665f7b-b08c-4ac3-9c18-ba926d44a60a"}) do
```
Comment (jalbertsr) [inline lib/modern_treasurer/counterparty/counter_party.ex:16]: same here

**Review by collegeimprovements — COMMENTED** [review]
code cleanup

**Review by jalbertsr — APPROVED** [review]

**Pull request merged**